src/SafeCurl.php

```diff
@@ -56,6 +56,18 @@ public function execute($url) {
         do {
             $url = $this->urlValidator->validateUrl($url);
 
+            if ($url && isset($url['host']) && isset($url['ips'])) {
+                $resolves = [];
+                foreach ($url['ips'] as $url_ip) {
+                    foreach ([80, 443, 8080] as $url_port) {
+                        $resolves[] = "{$url['host']}:$url_port:$url_ip";
+                    }
+                }
+                curl_setopt($this->curlHandle, CURLOPT_RESOLVE, $resolves);
+            } else {
+                throw new Exception("Cannot resolve host.");
```

Comment:
It looks like we already have an exception condition for this in `UrlValidator`.

https://github.com/vanilla/safecurl/blob/84be09328597549260d28a0b96db0bb30f58e25a/src/UrlValidator.php#L162

If `validateUrl` doesn't throw any exceptions, I think it should give us everything we need. If any of the pieces are missing or invalid, exceptions are thrown as part of validating the URL.